### ISSUE TITLE
Adding Draggable Markers

### ISSUE THREAD

**Issue by winecraft** [jmelosegui/GooglemapMvc#45]
Hello, I can see that you let us drag the map. However in admin mode, can you share some information on how
 -      Add a new icon/ with a template item or info box
 -      Allow user option to - drag the marker around showing the Icon location with confirmation / or turn off dragging.
 -       I wanted to add this code, but not break your build :)

Here is some js from google api to drag them around - how can we let the user save these positions back at the controller in ASP MVC
```javascript
marker = new google.maps.Marker(
{
    map:map,
    draggable:true,
    animation: google.maps.Animation.DROP,
    position: results[0].geometry.location
});
google.maps.event.addListener(marker, 'dragend', function() 
{
    geocodePosition(marker.getPosition());
});

function geocodePosition(pos) 
{
   geocoder = new google.maps.Geocoder();
   geocoder.geocode
    ({
        latLng: pos
    }, 
        function(results, status) 
        {
            if (status == google.maps.GeocoderStatus.OK) 
            {
                $("#mapSearchInput").val(results[0].formatted_address);
                $("#mapErrorMsg").hide(100);
            } 
            else 
            {
                $("#mapErrorMsg").html('Cannot determine address at this location.'+status).show(100);
            }
        }
    );
}
```

**Comment by jmelosegui** [comment]
Hi @winecraft,

I don't think I fully understand what you are trying to accomplish. 
I don't know what you mean by "admin mode" either.

If you want to know how to enable the drag feature in a marker and do something on 'dragend' you can see that in action in the following sample

[http://www.jmelosegui.com/map/marker/ClientEvents](http://www.jmelosegui.com/map/marker/ClientEvents)

Let me know if that is what you are looking for.

**Comment by winecraft** [comment]
Hi @jmelosegui, frist great job sir on your work. you're pure extragenius :) this is helping with my class work greatly... and understanding ASP MVC

I am trying to capture the lat/long co-ordinates of the draggable marker to where its finally moved to - i.e. dragend/stop event. I don't know how to get this information on the server side.
-  So, in the draggable marker - I was looking at the clientside events. Its hard to follow the flow because you have clientsideEvents with texwriter.writer - I tried to follow code, but could only find highlight events and was very confused by then :P  Can you please explain how to get the co-ordinates on the server side, with some comments on whats going on the clientevents Javascript and what its connecting backto on the server side.

-  http://bit.ly/1eWSbr4 Toggle/resize map to full screen or back down to widget size. Can you also tell me how to make it responsive, i.e. how to let user select: *view/fit the map to full page, or resize down to box layout.*

- Mapmarkers: is this the icon class, or an icon collection? - I'm asking because I want to be able to change the icon type and manage the icons. Where/which class would I use to change the icons?

- Overlay, Infowindow: which of this is the content window where I populate the content to display marker info on hover or click?

**Comment by winecraft** [comment]
Any help?

**Comment by jmelosegui** [comment]
Lot of question in just one issue. I will try to be concise :)

*_First Point_*

To capture the coordinates where the marker is dropped you will need to subscribe to the marker client envet "dropend"

```
 @(
            Html.GoogleMap()
                .Name("map")
                .Height(420)
                .Markers(m =>
                {
                    m.Add().Id("UniqueId-1")
                        .Title("Hello Marker Client Events!")
                        .Longitude(-82)
                        .Latitude(23)
                        .Draggable((bool)ViewData["draggable"])
                        );
                })
                .MarkerEvents(events => events.OnMarkerDragEnd("onDragEnd"))
        )
```

Then in the event handler you need to send those coordenates back to the server. 
To accomplish that you should use a client side framework to help you on the task (Ex: jQuery).

```
<script type="text/javascript">
        function onDragEnd(args) {
            $.ajax({
                url: "/{YourController}/{YourAction}",
                type: "POST",
                data: JSON.stringify({ 'latLng': args.latLng}),
                dataType: "json",
                traditional: true,
                contentType: "application/json; charset=utf-8",
                success: function (data) {                   
                    alert("Done");                   
                },
                error: function () {
                    alert("An error has occured!!!");
                }
            });
        }
    </script>
 ```
 
 *_Second Point_*
 
 The Second Point is not specifically related with my control or google map apis.
 If you want to use the browser fullscreen api you could read mor about it [here](https://developer.mozilla.org/en-US/docs/Web/Guide/API/DOM/Using_full_screen_mode)
 
 *_Third Point_*
 
 To accomplish that you should follow this [sample](http://www.jmelosegui.com/map/marker/CustomIcons)
 
 *_Fourth Point_*
 
  To accomplish that you should follow this [sample](http://www.jmelosegui.com/map/marker/InfoWindow)
